File: nightlife_engine/formatters/promotion_formatter.py
"""Promotion content formatter."""

from typing import Dict, Any
from .base_formatter import BaseFormatter


class PromotionFormatter(BaseFormatter):
    """
    Formats content for Promotions.

    Output format:
    • Headline
    • Promo copy (2–3 short sentences)
    • CTA line
    """

    def format(self, generated_content: Dict[str, Any]) -> str:
        """Format Promotion content."""
        headline = generated_content.get("headline", "")
        promo_copy = generated_content.get("promo_copy", "")
        cta = generated_content.get("cta", "")

        # Clean all parts
        headline = self._clean_output(headline)
        promo_copy = self._clean_output(promo_copy)
        cta = self._clean_output(cta)

        # Validate promo copy is 2-3 sentences
        sentence_count = promo_copy.count('.') + promo_copy.count('!') + promo_copy.count('?')
        if sentence_count < 2 or sentence_count > 3:
            raise ValueError(
                f"Promo copy must be 2-3 sentences (got {sentence_count})"
            )

        # Format output
        output_parts = [
            headline,
            "",
            promo_copy,
            "",
            cta
        ]

        return "\n".join(output_parts)

File: webapp/lib/formatters/promotion-formatter.ts
/**
 * Promotion content formatter
 */

import { BaseFormatter, GeneratedContent } from "./base-formatter";

export class PromotionFormatter extends BaseFormatter {
  format(generatedContent: GeneratedContent): string {
    const headline = generatedContent.headline as string || "";
    const promoCopy = generatedContent.promoCopy as string || "";
    const cta = generatedContent.cta as string || "";

    // Clean all parts
    const cleanedHeadline = this.cleanOutput(headline);
    const cleanedPromoCopy = this.cleanOutput(promoCopy);
    const cleanedCta = this.cleanOutput(cta);

    // Validate promo copy is 2-3 sentences
    const sentenceCount =
      (cleanedPromoCopy.match(/[.!?]/g) || []).length;
    if (sentenceCount < 2 || sentenceCount > 3) {
      throw new Error(
        `Promo copy must be 2-3 sentences (got ${sentenceCount})`
      );
    }

    // Format output
    return [cleanedHeadline, "", cleanedPromoCopy, "", cleanedCta].join("\n");
  }
}
